util-flywayinstaller/src/main/java/com/github/cafapi/util/flywayinstaller/Migrator.java

```diff
@@ -0,0 +1,117 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.util.flywayinstaller;
+
+import com.github.cafapi.util.flywayinstaller.exceptions.InvalidConnectionStringException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.flywaydb.core.Flyway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Migrator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(Migrator.class);
+    private static final String CONNECTION_URL_REGEX = "^.*\\/\\/.+:\\d+\\/";
+
+    private Migrator()
+    {
+    }
+
+    public static void migrate(
+        final boolean allowDBDeletion,
+        final String connectionString,
+        final String dbName,
+        final String username,
+        final String password
+    ) throws InvalidConnectionStringException, SQLException
+    {
+        LOGGER.debug("Arguments received"
+            + " allowDBDeletion: {}\n"
+            + " connectionString: {}\n"
+            + " dbName: {}\n"
+            + " username: {}\n"
+            + " password: {}", allowDBDeletion, connectionString, dbName, username, password);
+
+        LOGGER.info("Starting migration ...");
+        try (final BasicDataSource dbSource = new BasicDataSource()) {
+            dbSource.setUrl(checkAndConvertConnectionUrl(connectionString));
+            dbSource.setUsername(username);
+            dbSource.setPassword(password);
+
+            final boolean exists = checkDBExists(dbSource, dbName);
+            LOGGER.debug("Database exists: {}", exists);
+            if (!exists || allowDBDeletion) {
+                resetOrCreateDatabase(dbSource, exists, dbName);
+            }
+
+            LOGGER.info("About to perform DB update.");
+            final Flyway flyway = Flyway.configure()
+                .dataSource(dbSource.getUrl() + dbName, username, password)
+                .baselineOnMigrate(true)
+                .load();
+            flyway.migrate();
+            flyway.validate();
+            LOGGER.info("DB update finished.");
+        }
+        LOGGER.info("Migration completed ...");
+    }
+
+    private static void resetOrCreateDatabase(
+        final BasicDataSource dbSource,
+        final boolean exists,
+        final String dbName) throws SQLException
+    {
+        final String savedUrl = dbSource.getUrl();
+        dbSource.setUrl(dbSource.getUrl() + "postgres");
+        LOGGER.debug(dbSource.getUrl());
+        try (final Connection connection = dbSource.getConnection();
+             final Statement statement = connection.createStatement()) {
+            if (exists) {
+                LOGGER.info("force deletion has been specified.\nDeleting database {}", dbName);
+                statement.executeUpdate("DROP DATABASE " + dbName);
```

Comment:
Looks like more SQL injection potential to me.

Reply:
Same as [here](https://github.com/CAFapi/util-flyway-installer/pull/1#discussion_r788097242) We could potentially use a regex for validation the database name using that [documentation](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS)

Reply:
Removed